retrofit/src/test/java/retrofit2/RequestBuilderTest.java

```diff
@@ -2486,6 +2486,45 @@
     }
   }
 
+  @Test public void multipartPartsShouldBeInOrder() throws IOException {
+    class Example {
+      @Multipart
+      @POST("/foo")
+      Call<ResponseBody> get(@Part("first") String data, @Part("second") String dataTwo, @Part("third") String dataThree) {
+        return null;
+      }
+    }
+    Request request = buildRequest(Example.class, "firstParam", "secondParam", "thirdParam");
+    MultipartBody body = (MultipartBody) request.body();
+
+    Buffer buffer = new Buffer();
+    body.writeTo(buffer);
+    String readBody = buffer.readUtf8();
+
+    assertThat(readBody.indexOf("firstParam")).isLessThan(readBody.indexOf("secondParam"));
+    assertThat(readBody.indexOf("secondParam")).isLessThan(readBody.indexOf("thirdParam"));
+  }
+
+  @Test public void multipartPartsShouldBeInOrderWithUrlParam() throws IOException {
```

Comment:
I only have added this second version to reflect a code sample provided in the #1846.
Come to think of it twice, it is not needed, will remove.